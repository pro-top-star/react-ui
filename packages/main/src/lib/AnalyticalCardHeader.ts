import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
