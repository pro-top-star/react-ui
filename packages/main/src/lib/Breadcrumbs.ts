import { Breadcrumbs } from '../components/Breadcrumbs';

export { Breadcrumbs };
