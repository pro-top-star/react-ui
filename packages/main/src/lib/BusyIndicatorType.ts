import { BusyIndicatorType } from '../enums/BusyIndicatorType';

export { BusyIndicatorType };
