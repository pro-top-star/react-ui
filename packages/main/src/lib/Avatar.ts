import { Avatar } from '../webComponents/Avatar';

export { Avatar };
