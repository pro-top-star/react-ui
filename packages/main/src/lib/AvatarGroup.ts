import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
