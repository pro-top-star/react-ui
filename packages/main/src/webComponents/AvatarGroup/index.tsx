import { AvatarGroupType } from '@ui5/webcomponents-react/lib/AvatarGroupType';
import { AvatarSize } from '@ui5/webcomponents-react/lib/AvatarSize';
import { withWebComponent, WithWebComponentPropTypes } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents/dist/AvatarGroup';
import { FC, ReactNode } from 'react';

export interface AvatarGroupPropTypes extends Omit<WithWebComponentPropTypes, 'onClick'> {
  /**
   * Defines predefined size of the `Avatar`.
   *
   * Available options are:
   *
   * *   `XS`
   * *   `S`
   * *   `M`
   * *   `L`
   * *   `XL`
   */
  avatarSize?: AvatarSize;
  /**
   * Defines the mode of the `AvatarGroup`.
   *
   * Available options are:
   *
   * *   `Group`
   * *   `Individual`
   */
  type?: AvatarGroupType;
  /**
   * Defines the items of the `AvatarGroup`. Use the `Avatar` component as an item.
   *
   * **Note:** The UX guidelines recommends using avatars with "Circle" shape. Moreover, if you use avatars with "Square" shape, there will be visual inconsistency as the built-in overflow action has "Circle" shape.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Fired when the `AvatarGroup` is activated either with a click/tap or by using the Enter or Space key.
   */
  onClick?: (event: CustomEvent<{ targetRef: ReactNode; overflowButtonClicked: boolean }>) => void;
}

/**
 * Displays a group of avatars arranged horizontally. It is useful to visually showcase a group of related avatars, such as, project team members or employees. The component allows you to display the avatars in different sizes, depending on your use case. The `AvatarGroup` component has two group types:
 *
 * *   `Group` type: The avatars are displayed as partially overlapped on top of each other and the entire group has one click/tap area.
 * *   `Individual` type: The avatars are displayed side-by-side and each avatar has its own click/tap area.
 *
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/AvatarGroup" target="_blank">UI5 Web Components Playground</a>
 */
const AvatarGroup: FC<AvatarGroupPropTypes> = withWebComponent<AvatarGroupPropTypes>(
  'ui5-avatar-group',
  ['avatarSize', 'type'],
  [],
  [],
  ['click']
);

AvatarGroup.displayName = 'AvatarGroup';

AvatarGroup.defaultProps = {
  avatarSize: AvatarSize.S,
  type: AvatarGroupType.Group
};

export { AvatarGroup };
