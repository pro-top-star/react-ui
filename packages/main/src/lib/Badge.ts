import { Badge } from '../webComponents/Badge';
import type { BadgePropTypes } from '../webComponents/Badge';

export { Badge };
export type { BadgePropTypes };
