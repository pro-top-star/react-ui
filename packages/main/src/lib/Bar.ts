import { Bar } from '../webComponents/Bar';
import type { BarPropTypes } from '../webComponents/Bar';

export { Bar };
export type { BarPropTypes };
