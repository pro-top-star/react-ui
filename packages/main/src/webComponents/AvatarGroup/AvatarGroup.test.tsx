import { render } from '@shared/tests';
import { AvatarGroup } from '@ui5/webcomponents-react/lib/AvatarGroup';
import React from 'react';

describe('AvatarGroup', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<AvatarGroup />);
    expect(asFragment()).toMatchSnapshot();
  });
});
