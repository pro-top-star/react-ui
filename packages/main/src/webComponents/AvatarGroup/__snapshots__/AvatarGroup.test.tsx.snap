// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-avatar-group
    avatar-size="S"
    type="Group"
    ui5-avatar-group=""
  />
</DocumentFragment>
`;
